src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -81,28 +86,146 @@ void VulnerabilityScannerFacade::start(
+            RocksDBQueue<std::string>(DELTA_EVENTS_QUEUE_PATH));
+        m_syncQueue = std::make_shared<Utils::SafeQueue<std::string, RocksDBQueue<std::string>>>(
+            RocksDBQueue<std::string>(SYNCHRONIZATION_EVENTS_QUEUE_PATH));
+
+        // Threads to retry skipped events
+        m_deltaThread = std::thread(
+            [scanOrchestrator, deltaQueue = m_deltaQueue, &stopRetryThread = m_stopRetryThread]()
+            {
+                while (!stopRetryThread.load())
+                {
+                    for (int i = 0; i < SLEEP_RETRY_THREADS_SEC; ++i)
+                    {
+                        std::this_thread::sleep_for(std::chrono::seconds(1));
+                        if (stopRetryThread.load())
+                        {
+                            break;
+                        }
+                    }
+
+                    while (!deltaQueue->empty() && !stopRetryThread.load())
+                    {
+                        std::shared_ptr<std::string> returnStr = deltaQueue->pop();
+
+                        flatbuffers::Verifier deltaVerifier(reinterpret_cast<const uint8_t*>(returnStr->c_str()),
+                                                            returnStr->size());
+
+                        if (SyscollectorDeltas::VerifyDeltaBuffer(deltaVerifier))
+                        {
+                            std::variant<const SyscollectorDeltas::Delta*, const SyscollectorSynchronization::SyncMsg*>
+                                data = SyscollectorDeltas::GetDelta(reinterpret_cast<const char*>(returnStr->c_str()));
+
+                            try
+                            {
+                                logDebug1(WM_VULNSCAN_LOGTAG, "Retry dbsync_packages event.");
+                                scanOrchestrator->run(data);
+                            }
+                            catch (const std::exception& e)
+                            {
+                                logWarn(WM_VULNSCAN_LOGTAG, "Discarded dbsync_packages event: %s", e.what());
+                            }
+                        }
+                    }
+                }
+            });
+
+        m_syncThread = std::thread(
+            [scanOrchestrator, syncQueue = m_syncQueue, &stopRetryThread = m_stopRetryThread]()
+            {
+                while (!stopRetryThread.load())
+                {
+                    for (int i = 0; i < SLEEP_RETRY_THREADS_SEC; ++i)
+                    {
+                        std::this_thread::sleep_for(std::chrono::seconds(1));
+                        if (stopRetryThread.load())
+                        {
+                            break;
+                        }
+                    }
+
+                    while (!syncQueue->empty() && !stopRetryThread.load())
+                    {
+                        std::shared_ptr<std::string> returnStr = syncQueue->pop();
```

Comment:
The `pop()` method could return `nullptr` during the shutdown and this would cause a segmentation fault below at `returnStr->c_str()`.

Reply:
done